bindings/rust/sys/build.rs

```diff
@@ -0,0 +1,349 @@
+            "uc_tlb_entry" => Some(String::from("TlbEntry")),
+            "uc_control_type" => Some(String::from("ControlType")),
+            "uc_context_content" => Some(String::from("ContextMode")),
+            "uc_tcg_op_code" => Some(String::from("TcgOpCode")),
+            "uc_tcg_op_flag" => Some(String::from("TcgOpFlag")),
+            _ => None,
+        }
+    }
+
+    fn enum_variant_name(
+        &self,
+        enum_name: Option<&str>,
+        original_variant_name: &str,
+        _variant_value: EnumVariantValue,
+    ) -> Option<String> {
+        if let Some(enum_name) = enum_name {
+            if enum_name.starts_with("enum uc_") {
+                // Prefix to strip from enum variant names
+                let prefix = match enum_name.strip_prefix("enum uc_").unwrap() {
+                    "query_type" => "UC_QUERY",
+                    "tlb_type" => "UC_TLB",
+                    "control_type" => "UC_CTL",
+                    "context_content" => "UC_CTL_CONTEXT",
+                    "err" => "UC_ERR",
+                    "mem_type" | "mem_region" => "UC_MEM",
+                    "arch" => "UC_ARCH",
+                    "mode" => "UC_MODE",
+                    "prot" => "UC_PROT",
+                    "hook_type" => "UC_HOOK",
+                    "x86_insn" => "UC_X86_INS",
+                    "tcg_op_code" => "UC_TCG_OP",
+                    "tcg_op_flag" => "UC_TCG_OP_FLAG",
+                    other => format!("UC_{}", other.to_uppercase()).leak(),
+                }
+                .to_string()
+                    + "_";
+
+                // Strip prefix
+                let mut fixed = original_variant_name
+                    .strip_prefix(&prefix)
+                    .map(str::to_uppercase);
+
+                // Special handling for numeric register names in PPC and MIPS
+                if (enum_name == "enum uc_ppc_reg" || enum_name == "enum uc_mips_reg")
+                    && fixed.as_ref().is_some_and(|s| s.parse::<u32>().is_ok())
+                {
+                    fixed = fixed.map(|s| format!("R{s}"));
+                }
+
+                // Special handling for CPU variants that start with a number
+                if enum_name.contains("cpu")
+                    && fixed
+                        .as_ref()
+                        .is_some_and(|s| s.chars().next().unwrap().is_ascii_digit())
+                {
+                    fixed = fixed.map(|s| format!("Model_{s}"));
+                }
+
+                // Special handling for mode values
+                if enum_name == "enum uc_mode" {
+                    fixed = fixed.map(|s| match s.as_str() {
+                        "16" | "32" | "64" => format!("MODE_{s}"),
+                        _ => s,
+                    });
+                }
+
+                return fixed;
+            }
+        }
+
+        None
+    }
+}
+
+fn generate_bindings() {
+    const HEADER_PATH: &str = "../../../include/unicorn/unicorn.h";
```

Comment:
It would be better to use `CARGO_MANIFEST_DIR`.

Reply:
Sure, though note that `CARGO_MANIFEST_DIR` is now `/path/to/unicorn/bindings/rust/sys` because it's a workspace and the crate itself lives here